src/verifiers/documentStatus/tokenRegistry/ethereumTokenRegistryStatus.ts

```diff
@@ -42,18 +44,37 @@ export const getTokenRegistry = (
         ]
       );
     return issuers[0].tokenRegistry;
-  } else {
-    const { proof } = getData(document);
-    if (proof.method !== "TOKEN_REGISTRY")
+  }
+
+  if (utils.isWrappedV3Document(document)) {
+    if (!document.openAttestationMetadata.proof.value)
       throw new CodedError(
-        "Cannot validate non-token registry documents",
-        OpenAttestationEthereumTokenRegistryStatusCode.INVALID_VALIDATION_METHOD,
+        "Token registry is undefined",
+        OpenAttestationEthereumTokenRegistryStatusCode.UNDEFINED_TOKEN_REGISTRY,
         OpenAttestationEthereumTokenRegistryStatusCode[
-          OpenAttestationEthereumTokenRegistryStatusCode.INVALID_VALIDATION_METHOD
+          OpenAttestationEthereumTokenRegistryStatusCode.UNDEFINED_TOKEN_REGISTRY
         ]
       );
-    return proof.value;
+    return document.openAttestationMetadata.proof.value;
   }
+
+  throw new CodedError(
+    `Unrecognized document version`,
+    OpenAttestationEthereumTokenRegistryStatusCode.UNRECOGNIZED_DOCUMENT,
+    OpenAttestationEthereumTokenRegistryStatusCode[OpenAttestationEthereumTokenRegistryStatusCode.UNRECOGNIZED_DOCUMENT]
+  );
+};
+
+const getMerkleRoot = (
```

Comment:
find it a bit strange that this isn't inside open-attestation

Reply:
yea probably can add as a util there, good idea

Reply:
https://www.pivotaltracker.com/story/show/176860629